Enable system power management for NXP mcxa153/156/266/346/366

ZhaoxiangJin opened the pull request:
1. Add power state nodes for mcxa153/156/266/346/366, including `sleep`, `deepsleep `and `powerdown`.
2. Add power management related peripheral nodes for mcxa153/156/266/346/366, including `spc`, `cmc`, `vbat `and `wuu`.
3. Add power management driver support for MCXA153/MCXA156/MCXA266/MCXA346/MCXA366.
4. Enable `power_mgmt_soc `test for NXP frdm_mcxa153/156/266/346/366 board.
5. Enable systick and disable ostimer, this enabled systick as system timer.

<details>
<summary>test logs</summary>

```
*** Booting Zephyr OS build v4.3.0-2952-ge855856a5d63 ***
Running TESTSUITE power_mgmt
===================================================================
START - test_pm_dummyinit
I: PM dummy single-thread test started for one cycle
I: About to enter light sleep
PM >
I: Wake from Light Sleep
PM <
I: PM sleep residency 0.013 seconds
I: About to enter deep Sleep
PM >
I: Wake from Deep Sleep
PM <
I: PM sleep residency 1.183 seconds
I: PM dummy single-thread completed
 PASS - test_pm_dummyinit in 1.217 seconds
===================================================================
START - test_pm_multithread
I: Thread task A init
I: Thread task B init
I: PM multi-thread test started for cycles: 2
I: Suspend...
I: About to enter light sleep
PM >
I: Wake from Light Sleep
PM <
I: PM sleep residency 0.013 seconds
I: Resume
I: Suspend...
I: About to enter deep sleep
PM >
I: Wake from Deep Sleep
PM <
I: PM sleep residency 1.184 seconds
I: Resume
I: Suspend...
I: About to enter light sleep
PM >
I: Wake from Light Sleep
PM <
I: PM sleep residency 0.013 seconds
I: Resume
I: Suspend...
I: About to enter deep sleep
PM >
I: Wake from Deep Sleep
PM <
I: PM sleep residency 1.183 seconds
I: Resume
I: PM multi-thread completed
I: PM state[0] entry counter 0

I: PM state[0] exit counter 0

I: PM state[1] entry counter 3

I: PM state[1] exit counter 3

I: PM state[2] entry counter 0

I: PM state[2] exit counter 0

I: PM state[3] entry counter 2

I: PM state[3] exit counter 2

I: PM state[4] entry counter 0

I: PM state[4] exit counter 0

I: PM state[5] entry counter 0

I: PM state[5] exit counter 0

I: PM state[6] entry counter 0

I: PM state[6] exit counter 0

 PASS - test_pm_multithread in 2.477 seconds
===================================================================
START - test_pm_singlethread
I: PM single-thread test started for cycles: 2
I: About to enter light sleep
PM >
I: Wake from Light Sleep
PM <
I: PM sleep residency 0.013 seconds
I: About to enter deep Sleep
PM >
I: Wake from Deep Sleep
PM <
I: PM sleep residency 1.184 seconds
I: About to enter light sleep
PM >
I: Wake from Light Sleep
PM <
I: PM sleep residency 0.014 seconds
I: About to enter deep Sleep
PM >
I: Wake from Deep Sleep
PM <
I: PM sleep residency 1.183 seconds
I: PM single-thread completed
I: PM state[0] entry counter 0

I: PM state[0] exit counter 0

I: PM state[1] entry counter 3

I: PM state[1] exit counter 3

I: PM state[2] entry counter 0

I: PM state[2] exit counter 0

I: PM state[3] entry counter 2

I: PM state[3] exit counter 2

I: PM state[4] entry counter 0

I: PM state[4] exit counter 0

I: PM state[5] entry counter 0

I: PM state[5] exit counter 0

I: PM state[6] entry counter 0

I: PM state[6] exit counter 0

 PASS - test_pm_singlethread in 2.465 seconds
===================================================================
TESTSUITE power_mgmt succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [power_mgmt]: pass = 3, fail = 0, skip = 0, total = 3 duration = 6.159 seconds
 - PASS - [power_mgmt.test_pm_dummyinit] duration = 1.217 seconds
 - PASS - [power_mgmt.test_pm_multithread] duration = 2.477 seconds
 - PASS - [power_mgmt.test_pm_singlethread] duration = 2.465 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL

```
</details>

ZhaoxiangJin committed:
dts: arm: nxp: mcxa344: Add power-states for mcxa344

1. Add power-states for mcxa344
2. Add PM related-nodes for mcxa344.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
dts: arm: nxp: mcxa: Add power state nodes for mcxa156

1. Add power state nodes for mcxa156, including
sleep, deepsleep and powerdown.

2. Add power management related peripheral nodes
for mcxa156, including spc, cmc, vbat and wuu.

3. Add 'zephyr,cortex-m-idle-timer' node in frdm_mcxa156.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
soc: nxp/mcxa: Add power management support for MCXA156

Add power management support for MCXA/MCXA156.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
tests: power_mgmt_soc: enable test for frdm_mcxa156

1. enable power_mgmt_soc test for NXP frdm_mcxa156 board.
2. enable systick and disable ostimer, this enabled systick
as system timer.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
dts: arm: nxp: mcxa: Add power state nodes for mcxa153

1. Add power state nodes for mcxa153, including
sleep, deepsleep and powerdown.

2. Add power management related peripheral nodes
for mcxa153, including spc, cmc, vbat and wuu.

3. Add 'zephyr,cortex-m-idle-timer' node in frdm_mcxa153.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
soc: nxp/mcxa: Add power management support for MCXA153

Add power management support for MCXA/MCXA153.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
tests: power_mgmt_soc: enable test for frdm_mcxa153

1. enable power_mgmt_soc test for NXP frdm_mcxa153 board.
2. enable systick and disable ostimer, this enabled systick
as system timer.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

prayK reviewed (approved):
Nice to see the MCXA156 PM!

butok commented on tests/subsys/pm/power_mgmt_soc/boards/frdm_mcxa153.overlay:
```diff
@@ -0,0 +1,19 @@
+&systick {
+	status = "okay";
+};
+
```
Why not enable the systick by default?

ZhaoxiangJin committed:
dts: arm: nxp: mcxa: Add power state nodes for mcxaxx6

mcxaxx6: mcxa266, mcxa346, mcxa366.

1. Add power state nodes for mcxaxx6, including
sleep, deepsleep and powerdown.

2. Add power management related peripheral nodes
for mcxaxx6, including spc, cmc, vbat and wuu.

3. Add 'zephyr,cortex-m-idle-timer' node in frdm_mcxaxx6.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
soc: nxp/mcxa: Add power management support for MCXAxx6

MCXAxx6: MCXA266, MCXA346, MCXA366

Add power management support for MCXA/MCXAxx6.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
tests: power_mgmt_soc: enable test for frdm_mcxaxx6 boards

frdm_mcxaxx6: frdm_mcxa266, frdm_mcxa346, frdm_mcxa366

1. enable power_mgmt_soc test for NXP frdm_mcxaxx6 board.
2. enable systick and disable ostimer, this enabled systick
as system timer.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>
